forum/qa-plugin/report-reason/q2apro-flag-reasons-layer.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+class qa_html_theme_layer extends qa_html_theme_base
+{
+    private $isLogged;
+    
+    public function head_script()
+    {      
+        qa_html_theme_base::head_script();
+
+        $this->isLogged = qa_is_logged_in(); 
+        
+        if ($this->isLogged && 'question' === $this->template) {
+            $this->output(
+                '
+                <script>
+                    const flagAjaxURL = "' . qa_path('ajaxflagger') . '";
+                    const flagQuestionid = ' . $this->content['q_view']['raw']['postid'] . ';
+                </script>
+            '
+            );
+
+            $this->output(
+                '
+                <script type="text/javascript" src="' . QA_HTML_THEME_LAYER_URLTOROOT . 'script.js"></script>
+                <link rel="stylesheet" href="' . QA_HTML_THEME_LAYER_URLTOROOT . 'styles.css">
+            '
+            );
+        }
+
+    }
+
+    public function q_view_buttons($q_view)
+    {
+        if ($this->isLogged && isset($q_view['form']['buttons']['flag'], $q_view['raw']['postid'])) {
+            $q_view['form']['buttons']['flag']['tags'] =
+                'data-postid="' . $q_view['raw']['postid'] . '" data-posttype="q" ';
+        }
+
+        qa_html_theme_base::q_view_buttons($q_view);
+    }
+
+    public function a_item_buttons($a_item)
+    {
+        if ($this->isLogged && isset($a_item['form']['buttons']['flag'], $a_item['raw']['postid'])) {
+            $a_item['form']['buttons']['flag']['tags'] =
+                'data-postid="' . $a_item['raw']['postid'] . '" data-posttype="a" ';
+        }
+
+        qa_html_theme_base::a_item_buttons($a_item);
+    }
+
+    public function c_item_buttons($c_item)
+    {
+        if ($this->isLogged && isset($c_item['form']['buttons']['flag'], $c_item['raw']['postid'])) {
+            $c_item['form']['buttons']['flag']['tags'] = 'data-postid="' . $c_item['raw']['postid'] . '" data-posttype="c" data-parentid="' . $c_item['raw']['parentid'] . '" ';
+        }
+
+        qa_html_theme_base::c_item_buttons($c_item);
+    }
+
+    public function body_hidden()
+    {
+        if ($this->isLogged && 'question' === $this->template) {
+            $this->output(
+                '
+            <div id="flagbox-popup">
+                <div id="flagbox-center">
+                    <div class="qa-flag-reasons-wrap">
+                        <h4>
+                          '
+                          . qa_lang('q2apro_flagreasons_lang/reason')
+                          . '
+                        </h4>
+                        ';
```

Comment:
To mi wysrywa PHPStorma - w ogóle, co to ma być? Odpalasz funkcję output a w niej chcesz odpalić w jakiś dziwny sposób najpierw fora, póśniej w nim dwa echa?

Reply:
Wiem, coś kiedyś źle zacommitowałem, ale mam już poprawkę; wrzucę dziś.